app/gui2/playwright.config.ts

```diff
@@ -1,112 +1,123 @@
+              // FIXME: `--disable-gpu` disables `backdrop-filter: blur`, which is not handled by
+              // the software (CPU) compositor. This SHOULD be fixed eventually, but this flag
+              // MUST stay as CI does not have a GPU.
+              '--disable-gpu',
+              // Fully disable GPU process.
+              '--disable-software-rasterizer',
+              // Disable text subpixel antialiasing.
+              '--font-render-hinting=none',
+              '--disable-skia-runtime-opts',
+              '--disable-system-font-check',
+              '--disable-font-subpixel-positioning',
+              '--disable-lcd-text',
+            ],
+          },
+        }),
   },
-
-  /* Configure projects for major browsers */
-  projects: [
-    {
-      name: 'chromium',
-      use: {
-        ...devices['Desktop Chrome'],
-      },
-    },
-    {
-      name: 'firefox',
-      use: {
-        ...devices['Desktop Firefox'],
-      },
-    },
-    {
-      name: 'webkit',
-      use: {
-        ...devices['Desktop Safari'],
-      },
-    },
-
-    /* Test against mobile viewports. */
-    // {
-    //   name: 'Mobile Chrome',
-    //   use: {
-    //     ...devices['Pixel 5'],
-    //   },
-    // },
-    // {
-    //   name: 'Mobile Safari',
-    //   use: {
-    //     ...devices['iPhone 12'],
-    //   },
-    // },
-
-    /* Test against branded browsers. */
-    // {
-    //   name: 'Microsoft Edge',
-    //   use: {
-    //     channel: 'msedge',
-    //   },
-    // },
-    // {
-    //   name: 'Google Chrome',
-    //   use: {
-    //     channel: 'chrome',
-    //   },
-    // },
-  ],
-
-  /* Folder for test artifacts such as screenshots, videos, traces, etc. */
-  // outputDir: 'test-results/',
-
-  /* Run your local dev server before starting the tests */
+  // projects: [
+  // {
+  //   name: 'chromium',
+  //   use: {
+  //     ...devices['Desktop Chrome'],
+  //   },
+  // },
+  // {
+  //   name: 'firefox',
+  //   use: {
+  //     ...devices['Desktop Firefox'],
+  //   },
+  // },
+  // {
+  //   name: 'webkit',
+  //   use: {
+  //     ...devices['Desktop Safari'],
+  //   },
+  // },
+  // {
+  //   name: 'Mobile Chrome',
+  //   use: {
+  //     ...devices['Pixel 5'],
+  //   },
+  // },
+  // {
+  //   name: 'Mobile Safari',
+  //   use: {
+  //     ...devices['iPhone 12'],
+  //   },
+  // },
+  // ],
   webServer: {
-    /**
-     * Use the dev server by default for faster feedback loop.
-     * Use the preview server on CI for more realistic testing.
-    Playwright will re-use the local server if there is already a dev-server running.
-     */
-    command: process.env.CI ? 'vite preview --port 5173' : 'vite dev',
+    // The command-line flag is required
+    command: process.env.CI ? 'E2E=true vite preview --port 5173' : 'E2E=true vite dev',
     port: 5173,
     reuseExistingServer: !process.env.CI,
   },
-}
+})
 
-export default config
+expect.extend({
+  async toExist(locator: Locator) {
```

Comment:
I'm not sure that we need a custom assertion here. Seems like `toBeVisible` should do the work just fine. When you want to allow more than one such element to be matched, you can pair it with `first`, as the docs suggest:
https://playwright.dev/docs/api/class-locatorassertions#locator-assertions-to-be-visible